### ISSUE TITLE
[a11y]: PageHeader actions are not defined in a labeled region

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#8049]
### Package

@carbon/ibm-products

### Package version

v2.72.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Action buttons are not defined with in a labelled region.

#### Recommended fix
For the benefit of screen reader users, define the action buttons within a labeled region, based on the purpose of the action buttons — use an appropriate label, use role region with ARIA label.

#### User impact
The absence of a defined landmark role along with appropriate label for the content sections within the page  affects users who rely on assistive technologies, such as screen readers, to navigate and understand the structure of a web page. Without a defined role, these users may encounter difficulty in quickly locating and accessing important header information, leading to a less efficient and potentially frustrating browsing experience.

### WCAG 2.1 Violation

1.3.1. Info & Relationships

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/components-pageheader--with-actions-toolbar

### Steps to reproduce

1. Use a screen reader to navigate to the actions bar (icon buttons) in the page header.

No region is announced or conveyed to the user.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

---

### Acceptance criteria
- [ ] Evaluate appropriate roles or regions necessary for the ActionBar[^1].
  - [ ] Add roles/labels to the PageHeader v1 ActionBar
  - [ ] Consider adding roles/labels to PageHeader v2 `${blockClass}__breadcrumb__actions`
  - [ ] Consider adding roles/labels to web components - `page-header` `${prefix}--page-header__breadcrumb__actions`

[^1]: Some possible roles to consider: [`list`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/list_role), [`group`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/group_role), [`toolbar`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tooltip_role) although there may be other more appropriate roles.
